src/Arcus.EventGrid.Publishing/EventGridPublisherBuilderResult.cs

```diff
@@ -25,7 +25,8 @@ internal class EventGridPublisherBuilderResult : IEventGridPublisherBuilderWithE
         /// <param name="authenticationKey">Authentication key for the custom Event Grid topic</param>
         /// <exception cref="ArgumentException">The topic endpoint must not be empty and is required</exception>
         /// <exception cref="ArgumentException">The authentication key must not be empty and is required</exception>
-        public EventGridPublisherBuilderResult(string topicEndpoint, string authenticationKey)
+        /// <exception cref="UriFormatException">The topic endpoint must be a HTTP endpoint.</exception>
+        public EventGridPublisherBuilderResult(Uri topicEndpoint, string authenticationKey)
```

Comment:
Given the class is internal we might want to change this ctor to internal?

Reply:
Yes!